app/controllers/concerns/api_orderable.rb

```diff
@@ -3,7 +3,15 @@
 module ApiOrderable
   extend ActiveSupport::Concern
 
-private
+  module ClassMethods
+    attr_reader :model_klass
+
+  private
+
+    def orderable(model_klass: nil)
+      @model_klass = model_klass
+    end
+  end
```

Comment:
Let's avoid unnecessary meta-programming. I'd rather the class to injected into the `#sort_params` method.

```ruby
def sort_params(params, model: controller_name.classify.constantize)
  # ...
end
```
and in the controller
```ruby
def ecf_partnerships
  @ecf_partnerships ||= ecf_partnerships_query.partnerships.order(sort_params(params, model: Api::V3::ECFPartnershipSerializer))
end
```